aten/src/ATen/native/native_functions.yaml

```diff
@@ -1153,6 +1153,13 @@
     CPU: dot
     CUDA: dot_cuda
 
+- func: vdot(Tensor self, Tensor tensor) -> Tensor
```

Comment:
"Tensor self, Tensor other" to be consistent with other binary PyTorch functions

Reply:
`dot` just above is using `Tensor self, Tensor tensor`. Also, what's the reasoning behind having `dot` as `function, method`, and `vdot` as just `function`?

Reply:
Ug. I guess that's OK although `dot`'s signature is also wrong.

Methods are nice (and maybe still required to be put into method_tests?).

Reply:
For this operator I avoided `method` for 2 reasons.

* https://numpy.org/doc/stable/reference/generated/numpy.vdot.html Signature of numpy doesn't have it.
* Second since it returns a 1-element tensor, using it in method chaining might lead to confusion. (Maybe overly cautious)

Do let me know if that makes or we should indeed have method variant.

Reply:
numpy does not have methods for most things, but pytorch does. `dot` is a method, so it would make sense for `vdot` to be a method too.

Reply:
Sure.